src/main/java/nl/tudelft/semgroup4/GameStateController.java

```diff
@@ -0,0 +1,93 @@
+package nl.tudelft.semgroup4;
+
+import nl.tudelft.model.Game;
+import nl.tudelft.semgroup4.logger.LogSeverity;
+import nl.tudelft.semgroup4.resources.ResourcesWrapper;
+
+import org.newdawn.slick.GameContainer;
+import org.newdawn.slick.Graphics;
+import org.newdawn.slick.Input;
+import org.newdawn.slick.SlickException;
+import org.newdawn.slick.state.StateBasedGame;
+
+public class GameStateController {
+    
+    private Game currentGame;
```

Comment:
This would cause a circular reference, which you want to avoid. It would also make it harder to test the code. 

Also, the methods in the gamestatecontroller can never be completely self contained, because some of the variables that are passed through only have method scope in GameState. So these would still need to be passed through the method. So I will leave the controller as is.